netket/sampler/base.py

```diff
@@ -77,11 +118,20 @@ def __post_init__(self):
                 + "instead, type {} is not.".format(type(self.hilbert))
             )
 
-        if not isinstance(self.n_chains, int) and self.n_chains >= 0:
+        if not isinstance(self.n_chains, int) or self.n_chains < 0:
             raise ValueError("n_chains must be a positivee integer")
 
-        # if not isinstance(self.machine_pow, int) and self.machine_pow>= 0:
-        #    raise ValueError("machine_pow must be a positivee integer")
+        if not isinstance(self.machine_pow, int) or self.machine_pow < 0:
+            raise ValueError("machine_pow must be a positivee integer")
```

Comment:
Since we're touching this line anyways...
```suggestion
            raise ValueError("machine_pow must be a positive integer")
```